src/providers/wechat.js

```diff
@@ -0,0 +1,30 @@
+export default function WeChat(options) {
+  return {
+    id: "wechat",
+
+    name: "WeChat",
+    type: "oauth",
+    version: "2.0",
+    scope: options.scope,
+    params: { grant_type: "authorization_code" },
+    authorizationParams: {
+      appid: options.clientId
+    },
+    protection: "state",
+
+    accessTokenUrl: "https://api.weixin.qq.com/sns/oauth2/access_token",
+    authorizationUrl:
+      "https://open.weixin.qq.com/connect/oauth2/authorize?response_type=code",
+    profileUrl: "https://api.weixin.qq.com/sns/userinfo",
+
+    async profile(profile) {
+      return {
+        id: profile.openid,
+        nickname: profile.nickname,
+        avatar: profile.headimgurl
```

Comment:
The upcoming version will require all built-in providers to return a unified output here by default, which is `id`, `name`, `email`, `image`, defaulting to `null`. Read #2411 and the corresponding RFC #1846 for context.